chart/values.yaml

```diff
@@ -2,13 +2,30 @@ image:
   repository: teamspeak
   tag: 3.12.0
   pullPolicy: Always
-  # Secret must be manually created in the namespace.
+  ## Secret must be manually created in the namespace.
   # pullSecret: "nexus-pull-secret"
 podAnnotations: {}
-#only NodePort and LoadBalancer supported, as this is UDP traffic
+## only NodePort and LoadBalancer supported, as this is UDP traffic
 service:
   type: LoadBalancer
+  ## ip requests a specific IP address from the LoadBalancer (Supportd only for LoadBalancer)
+  #ip: "IPAddress"
   nodePort: 30987
+  ## tcp.enabled creates the TCP service if true (supported only if your LB implementation suports both TCP and UDP)
+  ## Needed until https://github.com/kubernetes/kubernetes/issues/23880 is implemented
+  ## See https://github.com/janosi/enhancements/blob/mixedprotocollb/keps/sig-network/20200103-mixed-protocol-lb.md#implementation-detailsnotesconstraints
+  ## Some mixed ports implementations use an annotation to merge services
+  ## known anotations:
+    ## MetalLB - metallb.universe.tf/allow-shared-ip: [string]
+    ## Azure CPI LB - service.beta.kubernetes.io/azure-load-balancer-mixed-protocols: [bool] (Not tested)
+  tcp:
+    enabled: false
+    ## type sets the implementation type. Values either combined | seperate. See docs.
+    type: "seperate"
```

Comment:
idea:
to remove complexity for the user, how about using a boolean (sth like, but better than "use-same-ip") and then automatically setting all known annotations to the service through templating directly?

Reply:
Hi,
i'd object that idea.
For cotext: I'm currently running a small cluster with metallb and only one floating ip, which metallb controls.
The `allow-shared-ip` annotation is used to control which services can possibly share one IP address, with all the sevices sharing one value of the annotation being allowed to share an IP address. Now if the annotation would be moved into templating completely, i would need to amend/change the template instead of only providing the correct annotation/value in the values.

P.S.: Thank you for providing this chart, first of all.

Reply:
I see - thanks for the contribution then :)